### PR TITLE
Add ability to inject conjur-access-token volume mounts

### PR DESCRIPTION
**Pull request by doodlesbykumbi**
When injecting the authenticator sidecar the app containers require access to the conjur-access-token volume that the authenticator writes the access token to. This commit provides a mechanism for injecting access token volume mounts as part of the sidecar injector process. An operator is able to specify the candidate containers by specifying their names as a comma separated string under the annotation `sidecar-injector.cyberark.com/conjurTokenReceivers`.

In this PR is also:
1. The first unit test ever in this repo. YAY!
2. Adding a flag for running the server without TLS
3. A CHANGELOG.md
4. A Github Actions pipeline for running the unit tests

Closes https://github.com/cyberark/sidecar-injector/issues/25

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -74,6 +74,9 @@ Injects:
   + a newly created shareable **in-memory Volume** named `conjur-access-token` used to store the access token 
   + a configurable **Authenticator container** with:
     + a **Volume Mount** at `/run/conjur` of the `conjur-access-token` Volume
+  + a read-only **Volume Mount** at `/run/conjur` of the `conjur-access-token` Volume for
+    each of the containers whose names appears in the comma-separated list of container
+    names that you get to configure.
```
Comment (izgeri) [inline README.md:79]:
```suggestion
    names that you may configure.
```

**Comment by izgeri** [comment]
Connected to #25

File: pkg/inject/authenticator-admission-request.json.tmpl [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{
```
Comment (hughsaunders) [inline pkg/inject/authenticator-admission-request.json.tmpl:1]: These look like test fixtures, should they go in a fixtures directory?

File: pkg/inject/server.go [hunk 3]
```diff
@@ -54,43 +55,55 @@ type WebhookServer struct {
...
 
 	injectType, err := getAnnotation(&pod.ObjectMeta, annotationInjectTypeKey)
 	containerMode, err := getAnnotation(&pod.ObjectMeta, annotationContainerModeKey)
 	containerName, err := getAnnotation(&pod.ObjectMeta, annotationContainerNameKey)
-
+	conjurTokenReceiversStr, err := getAnnotation(
+		&pod.ObjectMeta,
+		annotationConjurTokenReceiversKey,
+	)
+	conjurTokenReceivers := strings.Split(conjurTokenReceiversStr, ",")
```
Comment (hughsaunders) [inline pkg/inject/server.go:103]: Allow a space after the comma
```suggestion
	conjurTokenReceivers := regexp.MustCompile(", ?").Split(conjurTokenReceiversStr, -1)
```

Comment (hughsaunders) [reply pkg/inject/server.go:103]: Hah, just realised this has no effect, because you're trimming off spaces anyway.

**Comment by doodlesbykumbi** [comment]
@izgeri Turns out there was no CHANGELOG. I've just added one

**Pull request merged**